VBA/Office-Shared-VBA/articles/about-autosave.md

```diff
@@ -0,0 +1,120 @@
+
+## Overview of AutoSave
+
+When a file is hosted in the cloud (that is, OneDrive, OneDrive for Business, or SharePoint Online), AutoSave enables the user's edits to be saved automatically and continuously. When the file is shared with others, then the others' changes will be merged into this user's version of the file. If AutoSave is turned off, then save must be triggered manually for the user's changes to be persisted in the cloud and for that user to receive others' changes.
+
+Currently, Excel, Word, and PowerPoint provide a **BeforeSave** event that allows a developer to execute code after the user triggers a save but before the save actually occurs. Excel also provides an **AfterSave** event that can execute macro or add-in code after the save completes.
+
+When AutoSave is enabled, these events will fire automatically on a periodic basis without user interaction. Because of this new behavior when AutoSave is enabled, add-ins and macros that leverage these events may experience problems when AutoSave is on.
+
+In general, these issues can be avoided if the user chooses to disable AutoSave. You can also do this on the user’s behalf using the **autoSaveOn** property (see the example below). You can also take steps as a developer to mitigate these potential problems so that your add-ins and macros work smoothly, even if AutoSave is enabled.
+
+## Potential issues with save events and AutoSave
+
+### Issue 1: Code in **BeforeSave** or **AfterSave** events runs too long
+
+#### Description
+
+In general, Word, Excel and PowerPoint are not responsive to user interaction while add-in or macro code is being run. Therefore, if your code in a BeforeSave or AfterSave event handler takes too long to run, it may significantly degrade the user experience. When AutoSave is disabled, this code is only run when the user explicitly chooses to save, so a delay is not as noticeable and can be avoided by the user until he or she is ready to save. When AutoSave is enabled, this code will run automatically on a periodic basis, which has the potential to interrupt the user, especially if the code takes a long time.
+
+#### Example Scenario
+
+Imagine an add-in that allows the user to create custom maps based on data in an Excel workbook. Such an add-in might have **BeforeSave** code that serializes any maps that the user has created and stores them in the workbook in a CustomXML part. This process might take a second to complete and Excel is unresponsive while this is happening. When AutoSave is off, the user gets to choose when he or she wants to save and therefore, even though the add-in slows down the save process slightly, the user does not notice. When AutoSave is enabled, this **BeforeSave** code is executed automatically on a periodic basis even if the user is in the middle of something else (like typing data into a cell) which could be extremely annoying.
```

Comment:
"and Excel is unresponsive while this is happening -> "and Excel **would be** responsive while this is happening"